### ISSUE TITLE
Letters change to a question mark

### ISSUE THREAD

**Issue by Hrommi** [BrowserSync/browser-sync#1523]
Hi. Sorry for my bad English.

### Issue details

Sometimes one letter is replaced by two question marks. This happens rarely in random places. But this is a big problem for my `gemini` tests.
![browsersync](https://user-images.githubusercontent.com/10770815/37602032-9261e5ec-2b9c-11e8-8fa6-24df0507a6bf.png)

### Please specify which version of Browsersync, node and npm you're running

- Browsersync 2.23.6
- Node v8.9.4
- Npm 5.6.0

### Affected platforms

- windows 10

### Browsersync use-case

-  Gulp

**Comment by shakyShane** [comment]
unfortunately Browsersync does not currently support anything other charset than utf8 (due to the way responses are modified) - so what you're seeing is likely because you're using something other than utf8.

I wish I had a better answer :(

**Comment by bfred-it** [comment]
Yeah the answer should be “use utf-8 because text encoding is a solved problem” 😅

**Comment by Hrommi** [comment]
But I use `<meta charset="utf-8">`.

**Comment by bfred-it** [comment]
Is the editor set to that encoding? Are they plain HTML files or are you using proxy?

**Comment by Hrommi** [comment]
Editor set UTF-8 without BOM. Open plain HTML files.
My settings for `browsersync`:
```
  server: {
    baseDir: './build'
  },
  port: 8080,
  notify: false,
  ghostMode: false
```

**Comment by acrolink** [comment]
I have the same issue and I am using `<meta charset="utf-8">`.
![2019-05-19 15_22_18-تصميم وإنشاء برامج محوسبة _ أكرولينك](https://user-images.githubusercontent.com/13219117/57982149-16c9c000-7a4a-11e9-8b89-fd4493ba8ae2.png)

**Comment by direc85** [comment]
I ran into this today: hit F5 a few times on a certain page, and one certrain character (Finnish ä) on the page may or may not turn into two question marks. The HTTP header reported different content-length accordingly, and the raw response differed only in that one spot.

My colleague finally found out about [this comment](https://github.com/BrowserSync/browser-sync/issues/377#issuecomment-407020407) which suggested that changing `<meta charset="utf-8">` into `<meta charset="utf-8" />` fixes the issue - and it did for me, too.

Node v14.17.5
NPM 6.14.14
browser-sync: 2.27.9

**Comment by mahnunchik** [comment]
Any news? I've faced with the same issue.

* macOS 12.4
* NodeJS v16.15.1

**Comment by direc85** [comment]
One weird thing that I noticed (soon after my previous post) that if I modify the page contents even the slightest (usually, it depends where the change is in the page), the error does not occur. I think that explains why the tag manipulation works in the linked issue, too, because it simply modifies the processed content (and/or changes the content length).

I don't think the page this occurred in for me surfaces this anymore (there were structural changes), but apparently the issue is still there. As it only happens _sometimes_ when smashing F5, it smells like a race condition to me, which are always very Fun™ to track down...

Unfortunately I'm unable to share the exact code that caused it. Could someone share their code, or better yet, come up with an example project that triggers this issue?

**Comment by emilsrits** [comment]
Happens completely randomly for me. Sometimes special characters get replaced by ��, sometimes they don't.
Noticed once that two pages that had identical text, on one it was malformed on other not.
